### PR TITLE
Add custom headers

### PR DESCRIPTION
**Pull request by SiebeVE**
This will allow us to request custom examples from the mock server

### CODE REVIEW

File: AbstractClient.php [hunk 1]
```diff
@@ -91,10 +118,20 @@ protected function performRequest(Endpoint\EndpointInterface $endpoint)
         return $body;
     }
 
+    protected function getUriFromEndpoint(EndpointInterface $endpoint): string
+    {
+        $uri = $endpoint->uri();
+        if ($this->prependPath !== null) {
+            return $this->prependPath . $uri;
```
Comment (arnedesmedt) [inline AbstractClient.php:125]: Couldn't we just add this to the server url? https://swagger.io/docs/specification/api-host-and-base-path/
In this case we change the base uri from our guzzle client?

File: AbstractClient.php [hunk 2]
```diff
@@ -124,6 +161,12 @@ protected function buildOptions(Endpoint\EndpointInterface $endpoint) : array
             $options['form_params'] = $formData;
         }
 
+        if (count($this->customHeaders) > 0) {
+            $options['headers'] = $this->customHeaders;
+
+            $this->customHeaders = [];
+        }
```
Comment (arnedesmedt) [inline AbstractClient.php:168]: Instead of introducing a new property I would use the optionBuilder here. But we should change it a bit, that it doesn't return if the $options !== null and add extra constraints to the if's above, so we don't override the query/json/form_params options if they are already set.

File: AbstractClient.php (hunk 1 above)
Comment (SiebeVE) [reply AbstractClient.php:125]: No, Guzzle is following the RFC standard for relative URI's: https://docs.guzzlephp.org/en/stable/quickstart.html#creating-a-client.
TLDR: sdk's have their paths with a slash, so all paths after the domain are ignored in the base uri.

File: AbstractClient.php (hunk 2 above)
Comment (SiebeVE) [reply AbstractClient.php:168]: Yes, that is the reason why I added a new property. I will refactor it a bit.

File: AbstractClient.php [hunk 3]
```diff
@@ -91,36 +103,41 @@ protected function performRequest(Endpoint\EndpointInterface $endpoint)
...
         }
 
         $params = $endpoint->params();
         $body = $endpoint->body();
         $formData = $endpoint->formData();
 
-        $options = [];
-
-        if (! empty($params)) {
+        if (! empty($params) && ! empty($options['query'])) {
```
Comment (arnedesmedt) [inline AbstractClient.php:132]: Isn't this a bit strange?

Comment (SiebeVE) [reply AbstractClient.php]: What exactly do you mean? I've refactored it a bit.

File: AbstractClient.php [hunk 4]
```diff
@@ -91,36 +103,41 @@ protected function performRequest(Endpoint\EndpointInterface $endpoint)
         return $body;
     }
 
+    protected function getUriFromEndpoint(EndpointInterface $endpoint): string
```
Comment (arnedesmedt) [inline AbstractClient.php:106]: I would rename this to uriWithPrependPath. It says much more wath the function is doing.

File: AbstractClient.php (hunk 3 above)
Comment (arnedesmedt) [reply AbstractClient.php]: Ok, I was a bit confused when I wrote this comment.

File: AbstractClient.php (hunk 4 above)
Comment (SiebeVE) [reply AbstractClient.php:106]: Done!
